### PR TITLE
Add description field to networks

### PR DESCRIPTION
**Pull request by albertodonato**
Add a description field to networks. It can be changed with `lxc network edit`

### CODE REVIEW

File: test/suites/network.sh [hunk 1]
```diff
@@ -15,6 +18,14 @@ test_network() {
   lxc network set lxdt$$ ipv6.dhcp.stateful true
   lxc network delete lxdt$$
 
+  # edit network description
+  config_file=$(mktemp)
+  lxc network create lxdt$$
+  lxc network show lxdt$$ | sed 's/^description:/description: foo/' > "$config_file"
+  lxc network edit lxdt$$ < "$config_file"
+  lxc network show lxdt$$ | grep -q 'description: foo'
+  lxc network delete lxdt$$
```
Comment (brauner) [inline test/suites/network.sh]: Can you please also add a test for `lxc network set` to change the current set description and `lxc network unset`?

Comment (albertodonato) [inline test/suites/network.sh]: AFAICS `lxc network [set|unset]` are meant to modify the network configuration, not any top-level network attribute (like name, description or type).

It's not currently possible to change the description through them.

Comment (brauner) [inline test/suites/network.sh]: Ah, was that the original intention? If so, cool.

**Comment by brauner** [comment]
Misses a `go fmt ./...`.

**Comment by albertodonato** [comment]
@brauner  yes, as mentioned in #2022 it's an edit-only property (at least for now). I think it's ok, given it's not something the use would frequently edit

File: test/suites/network.sh (hunk 1 above)
Comment (stgraber) [inline test/suites/network.sh]: Yes, that's the behavior we have for all the others. "edit" will usually let you modify it.

**Comment by stgraber** [comment]
Cool, so I agree with @brauner's comment on IRC. We should have this pull request do all of them. Ideally with 1 commit per object getting the new description field (including updates to doc/rest-api.md as needed), followed by a commit which adds the API extension and doc/api-extensions.md entry.

I'll mark this branch incomplete for now and we'll merge once we have the other bits too.

File: test/suites/network.sh [hunk 2]
```diff
@@ -15,6 +18,14 @@ test_network() {
   lxc network set lxdt$$ ipv6.dhcp.stateful true
   lxc network delete lxdt$$
 
+  # edit network description
+  config_file=$(mktemp)
+  lxc network create lxdt$$
+  lxc network show lxdt$$ | sed 's/^description:/description: foo/' > "$config_file"
+  lxc network edit lxdt$$ < "$config_file"
```
Comment (stgraber) [inline test/suites/network.sh]: Note that you can just pipe directly from show into edit, no need to use a temporary file

**Comment by stgraber** [comment]
One extra thing. I just noticed that the "description" field on image aliases is "VARCHAR(255)". We probably want to bump it to "TEXT" for consistency as part of this.

**Pull request closed without merging**